compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -723,11 +729,23 @@ impl<'hir> LoweringContext<'_, 'hir> {
         let trait_item_def_id = hir_id.expect_owner();
 
         let (generics, kind, has_default) = match &i.kind {
-            AssocItemKind::Const(box ConstItem { ty, expr, .. }) => {
-                let ty =
-                    self.lower_ty(ty, &ImplTraitContext::Disallowed(ImplTraitPosition::ConstTy));
-                let body = expr.as_ref().map(|x| self.lower_const_body(i.span, Some(x)));
-                (hir::Generics::empty(), hir::TraitItemKind::Const(ty, body), body.is_some())
+            AssocItemKind::Const(box ConstItem { generics, ty, expr, .. }) => {
+                let (generics, kind) = self.lower_generics(
+                    &generics,
+                    Const::No,
```

Comment:
@fee1-dead (since you're already here), Fyi, I've just thoughtlessly added `Const::No` here and in similar places since I haven't looked into `#![feature(effects)]` at all yet. Either you or I can fix this in a follow-up if it needs fixing.

Reply:
Yeah this should be fine. `Const::Yes` should be passed only if this item needs host effect param (i.e. like const fn where it can be either const or runtime) Const items don't need those since the host effect is always off